### DISCUSSION TITLE
如何通过代码的方式设置某列的默认过滤条件？

### DISCUSSION THREAD

**Discussion by SeptSkyTime** [dotnetcore/BootstrapBlazor#3257] [Q&A] [answered]
在官网的表格示例代码中，有类似的例子，然而在我的代码中执行时，发现column.Filter为null，此时该如何通过代码创建Filter的实例？
我尝试创建Filter及其FilterAction的实例，无法正常运行，我的代码如下：

        private async Task SetDefaultFilterInCode(long parentId)
        {
            var column = tableObjectRef.Columns.First(x => x.GetFieldName() == "ParentTopicId");

            // 创建过滤条件
            var filter = new FilterKeyValueAction()
            {
                FieldValue = parentId,
                FilterAction = FilterAction.Equal
            };

            // 如何为column.Filter创建实例，我下面的代码无法达到预期效果
            if (column.Filter == null)
            {
                column.Filter = new TableFilter();
                if (column.Filter.FilterAction == null)
                    column.Filter.FilterAction = new LookupFilter();
            }
            var filterAction = column.Filter?.FilterAction;
            if (filterAction != null)
                await filterAction.SetFilterConditionsAsync(filter);
        }
谢谢！

**Comment by ArgoZhang** [comment]
@SeptSkyTime 代码太少，太片面，根本无法推断问题点在哪里。可以提供一个仓库复现问题。

**Reply by SeptSkyTime** [reply, marked as answer]
> @SeptSkyTime 代码太少，太片面，根本无法推断问题点在哪里。可以提供一个仓库复现问题。

我通过对比验证：
    如果在按钮事件中调用SetDefaultFilterInCode方法，column.Filter不为null，如果在页面加载事件OnAfterRenderAsync中，在firstRender为true时调用该方法，column.Filter为null。改为firstRender为false时调用，就可以了。
感谢提示！
